backend/open_webui/internal/pocketbase/REPOSITORIES_AND_ADAPTERS.md

```diff
@@ -0,0 +1,45 @@
+# Repository Interfaces and PocketBase Adapters
+
+This document defines repository interfaces for Open WebUI domain models and the mapping from current SQLAlchemy-backed methods to PocketBase (PB) implementations using the Python client (`vaphes/pocketbase`).
+
+## Strategy (explicit)
+- [x] Do NOT emulate `SessionLocal()` / `get_db` for PB.
+  - We will not duck-type SQLAlchemy sessions; instead, we will implement PB adapters and update the existing model modules to call them inside their methods, preserving public signatures.
+- [x] No per-model feature flags in production. A single backend is targeted post-migration (PB).
+
+## Principles
+- Keep existing API shapes returned to the rest of the app unchanged.
+- Repositories encapsulate all persistence details (SQL or PB) and mapping between PB records and Pydantic models.
+- PB adapters must be idempotent and handle retries.
```

Comment:
Consider if retries are better added to [vaphes/pocketbase](https://github.com/vaphes/pocketbase), or are already there.